packages/react-hooks/src/ConnectionProvider.tsx

```diff
@@ -71,10 +76,26 @@ const ConnectionProvider: React.FC<Props> = ({ agent, children }) => {
           connections: newConnectionsState,
         })
       }
-      agent?.events.on(ConnectionEventTypes.ConnectionStateChanged, listener)
+
+      const deletedListener = async (event: RecordDeletedEvent<ConnectionRecord>) => {
+        if (event.payload.record.type !== ConnectionRecord.type) {
+          return
+        }
+        const newConnectionsState = [
+          ...connectionState.connections.filter((connection) => connection.id != event.payload.record.id),
+        ]
```

Comment:
filter already returns a new array
```suggestion
        const newConnectionsState = connectionState.connections.filter((connection) => connection.id != event.payload.record.id)
```